quic/s2n-quic-transport/src/connection/connection_impl.rs

```diff
@@ -1656,6 +1656,24 @@ impl<Config: endpoint::Config> connection::Trait for ConnectionImpl<Config> {
         self.accept_state = AcceptState::Active;
     }
 
+    // Translates protocol errors into more useful user errors
+    fn translate_connection_err(&self, error: connection::Error) -> Result<(), connection::Error> {
```

Comment:
I don't think this is the right place for this kind of thing. It's probably better to put it in [connection::Error](https://github.com/aws/s2n-quic/blob/e4314d2763f99edb79326baf925f88b89cd4f78c/quic/s2n-quic-core/src/connection/error.rs#L159). I would also call it something like `into_accept_error` and hide it like the other methods in there: https://github.com/aws/s2n-quic/blob/e4314d2763f99edb79326baf925f88b89cd4f78c/quic/s2n-quic-core/src/connection/error.rs#L279

Reply:
Also we should call this function from the stream manager so we're not copy/pasting this logic everywhere